tests/test_policy.py

```diff
@@ -1336,6 +1336,23 @@ def test_env_var_extension(self):
         p.conditions.env_vars['account'] = {'name': 'mickey'}
         self.assertFalse(p.is_runnable())
 
+    def test_env_var_extension_with_expand_variables(self):
+        p = self.load_policy({
+            'name': 'profx',
+            'resource': 'aws.ec2',
+            'description': 'Test var extension {var1}',
+            'conditions': [{
+                'type': 'value',
+                'key': 'account.name',
+                'value': 'deputy'}]})
+        p.expand_variables({"var1":"value1"})
+        p.validate()
+        self.assertEqual("Test var extension value1", p.data["description"])
+        p.conditions.env_vars['account'] = {'name': 'deputy'}
```

Comment:
expand variables is done after env_vars set, test should mirror that reproduce original failure, reference
https://github.com/cloud-custodian/cloud-custodian/blob/main/tools/c7n_org/c7n_org/cli.py#L619

Reply:
@kapilt @kentnsw btw: Where do I find the official docs about the `expand-variables` functionality and docs about how to pass external vars from c7n/c7n-orq to c7n-Policies for interpolation? I thought only a few core vars like {account_id} or {op} for example will be interpolated internally?

Reply:
@kapilt  right, I must be sleeping when I wrote that test case.

@dschro-1993 I found this section (https://cloudcustodian.io/docs/tools/c7n-org.html#defining-and-using-variables). But perhaps it's not detail enough to meet your expectation. My understanding is the `expend-variables` will use any vars defined in accounts.yml (when using c7n-org) plus the following:

[policy.py](https://github.com/cloud-custodian/cloud-custodian/blob/0035b92e9a8d4c0fb8dc8277616cc34ab14b175d/c7n/policy.py#L1227-L1253)
```python
        variables.update({
            # standard runtime variables for interpolation
            'account': '{account}',
            'account_id': self.options.account_id,
            'partition': partition,
            'region': self.options.region,
            # non-standard runtime variables from local filter/action vocabularies
            #
            # notify action
            'policy': self.data,
            'event': '{event}',
            # mark for op action
            'op': '{op}',
            'action_date': '{action_date}',
            # tag action pyformat-date handling
            'now': utils.FormatDate(datetime.utcnow()),
            # account increase limit action
            'service': '{service}',
            # s3 set logging action :-( see if we can revisit this one.
            'bucket_region': '{bucket_region}',
            'bucket_name': '{bucket_name}',
            'source_bucket_name': '{source_bucket_name}',
            'source_bucket_region': '{source_bucket_region}',
            'target_bucket_name': '{target_bucket_name}',
            'target_prefix': '{target_prefix}',
            'LoadBalancerName': '{LoadBalancerName}'
        })
```

Reply:
@kentnsw That's awesome, thanks for the link! Was not 100% aware of that functionality.